stream.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/google/uuid"
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/zerolog"
+	tmrpcclient "github.com/tendermint/tendermint/rpc/client"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+	tmjsonclient "github.com/tendermint/tendermint/rpc/jsonrpc/client"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+var (
+	started                = false
+	queryEventBankTransfer = fmt.Sprintf(
+		"%s='%s' AND %s='%s'",
+		tmtypes.EventTypeKey,
+		tmtypes.EventTxValue,
+		"message.action",
+		"/cosmos.bank.v1beta1.MsgSend",
+	)
+	queryInterval = 100 * time.Millisecond
+)
+
+type StreamCollector struct {
+	rpcClient          tmrpcclient.Client
+	logger             zerolog.Logger
+	recentlySeenHashes *simplelru.LRU
+	histogram          *prometheus.HistogramVec
+}
+
+func NewStreamCollector(tmRPC string, logger zerolog.Logger) (*StreamCollector, error) {
+	httpClient, err := tmjsonclient.DefaultHTTPClient(tmRPC)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient.Timeout = 500 * time.Millisecond
+
+	rpcClient, err := rpchttp.NewWithClient(tmRPC, httpClient)
+	if err != nil {
+		return nil, err
+	}
+	recentlySeen, err := simplelru.NewLRU(10, func(k interface{}, v interface{}) { /* noop */ })
+	if err != nil {
+		return nil, err
+	}
+	transfersValueHistogram := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "cosmos_bank_transfer_amount",
+			Help:    "Number of tokens transferred in a transfer message",
+			Buckets: prometheus.ExponentialBuckets(1e6, 10, 10),
+		},
+		[]string{"denom"},
+	)
+	return &StreamCollector{
+		rpcClient:          rpcClient,
+		logger:             logger,
+		recentlySeenHashes: recentlySeen,
+		histogram:          transfersValueHistogram,
+	}, nil
+}
+
+func (s StreamCollector) Start(
+	ctx context.Context,
+) error {
+	if !started {
```

Comment:
Whats the purpose of `started`?  Why would `Start` be called multiple times?

Reply:
I copied this from the price feeder which had done it this way, i think youre right that this wouldnt be called multiple times so we can remove